test/test_oneview_storage_volume_template.py

```diff
@@ -14,270 +14,187 @@
+    """
 
-        mock_ansible_instance.fail_json.assert_called_once_with(
-            msg=FAKE_MSG_ERROR
-        )
+    def setUp(self):
+        self.configure_mocks(self, StorageVolumeTemplateModule)
+        self.resource = self.mock_ov_client.storage_volume_templates
 
-    @mock.patch.object(OneViewClient, 'from_json_file')
-    @mock.patch('oneview_storage_volume_template.AnsibleModule')
-    def test_should_fail_when_update_raises_exception(self, mock_ansible_module, mock_ov_client_from_json_file):
-        mock_ov_instance = mock.Mock()
-        mock_ov_instance.storage_volume_templates.get_by.return_value = [DICT_DEFAULT_STORAGE_VOLUME_TEMPLATE]
-        mock_ov_instance.storage_volume_templates.update.side_effect = Exception(FAKE_MSG_ERROR)
+    def test_should_fail_when_create_raises_exception(self):
+        self.resource.get_by.return_value = []
+        self.resource.create.side_effect = Exception(FAKE_MSG_ERROR)
 
-        mock_ov_client_from_json_file.return_value = mock_ov_instance
-        mock_ansible_instance = create_ansible_mock_yaml(YAML_STORAGE_VOLUME_TEMPLATE_CHANGE)
-        mock_ansible_module.return_value = mock_ansible_instance
+        self.mock_ansible_module.params = PARAMS_FOR_PRESENT
 
         self.assertRaises(Exception, StorageVolumeTemplateModule().run())
 
-        mock_ansible_instance.fail_json.assert_called_once_with(
+        self.mock_ansible_module.fail_json.assert_called_once_with(
             msg=FAKE_MSG_ERROR
         )
 
-    @mock.patch.object(OneViewClient, 'from_json_file')
-    @mock.patch('oneview_storage_volume_template.AnsibleModule')
-    def test_should_fail_when_delete_raises_exception(self, mock_ansible_module, mock_ov_client_from_json_file):
-        mock_ov_instance = mock.Mock()
-        mock_ov_instance.storage_volume_templates.get_by.return_value = [DICT_DEFAULT_STORAGE_VOLUME_TEMPLATE]
-        mock_ov_instance.storage_volume_templates.delete.side_effect = Exception(FAKE_MSG_ERROR)
+    def test_should_fail_when_update_raises_exception(self):
+        self.resource.get_by.return_value = [STORAGE_VOLUME_TEMPLATE]
+        self.resource.update.side_effect = Exception(FAKE_MSG_ERROR)
 
-        mock_ov_client_from_json_file.return_value = mock_ov_instance
-        mock_ansible_instance = create_ansible_mock_yaml(YAML_STORAGE_VOLUME_TEMPLATE_ABSENT)
-        mock_ansible_module.return_value = mock_ansible_instance
+        self.mock_ansible_module.params = PARAMS_WITH_CHANGES
 
         self.assertRaises(Exception, StorageVolumeTemplateModule().run())
 
-        mock_ansible_instance.fail_json.assert_called_once_with(
+        self.mock_ansible_module.fail_json.assert_called_once_with(
             msg=FAKE_MSG_ERROR
         )
 
-    @mock.patch.object(OneViewClient, 'from_json_file')
-    @mock.patch('oneview_storage_volume_template.AnsibleModule')
-    def test_should_raise_exception_when_key_is_missing(self, mock_ansible_module, mock_ov_client_from_json_file):
```

Comment:
Why was the method `test_should_raise_exception_when_key_is_missing` removed?
The removed test covers this block of code from the module:

```python
             if not data.get('name'):
                  raise Exception(STORAGE_VOLUME_TEMPLATE_MANDATORY_FIELD_MISSING)
```